exchange/bidder.go

```diff
@@ -255,13 +272,13 @@ func getNativeImpByImpID(impID string, request *openrtb.BidRequest) (*openrtb.Na
 	return nil, errors.New("Could not find native imp")
 }
 
-func getAssetByID(id int64, assets []nativeRequests.Asset) nativeRequests.Asset {
+func getAssetByID(id int64, assets []nativeRequests.Asset) (*nativeRequests.Asset, error) {
 	for _, asset := range assets {
 		if id == asset.ID {
-			return asset
+			return &asset, nil
```

Comment:
Curious. Why the switch to a pointer?

Reply:
This function was initially not returning an error and therefore I changed that to a pointer to be able to do a nil check where this function is called but then realized that it's better to return an error and ended up keeping it as a pointer as it doesn't matter much either way in this case.

Nevertheless have changed it back and updated the PR.